### PR TITLE
Add templates

### LINKED ISSUE #3477: Create documentation templates for new topics
**Describe the change you'd like to see**

We should provide docs templates for
- new topics in the nav (template dir)
- 3 main types of content; procedures, conceptual and reference

**Additional context**
Hopefully having copy/paste/edit templates for docs will make it easier for eng and others to contribute to docs since it takes some of the guess work out of formatting and how to add new topics. There's sort of a dependency here on having better information architecture however this will be easier to clean up later IMO if the docs are at least in a consistent format.

**TO DO**

- [ ] Add new templates to ignored files in website as per https://github.com/knative/website/pull/291/files so that they don't get published
- [ ] Add templates for 3 main content types (can be iterated later)

**Out of scope initially**

- Guidance on doc weights and other frontmatter; use best judgement / ask in Slack - @omerbensaadon can you please make sure there are good contrib docs for dealing with frontmatter?

### PR DESCRIPTION
**Pull request by RichardJJG**

### CODE REVIEW

File: template-procedure.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+!!! warning
+    This is a warning.
+
+
+#### Markdown Embedded Image
+
+The following is an embedded image reference in markdown.
+
+![Annotated procedure topic](../image/example-reference-topic.png)
```
Comment (snneji) [inline template-procedure.md:212]: Could you include this image in the PR? I can't see it as one of the committed files.

File: template-procedure.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+
+#### Tabs
+
+Place multiple versions of the same procedure (such as different versions for different products)
+within tabs. Indent the opening tabs tags 3 spaces to make the tabs display properly.
+
+   {{< tabs name="name" default="name" >}}
```
Comment (snneji) [inline template-procedure.md:220]: The formatting for tabs has changed since the move to MKdocs. I've found that the content must be indented 4 spaces relative to the indentation for the `=== "tab name"` line, otherwise the tabs will break in various interesting ways. Also, there is no closing tag.

Formatting is as follows:

```
=== "tab1 name"

    Tab content

=== "tab2 name"

    Tab content
```

File: template-procedure.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+#### Markdown Embedded Image
+
+The following is an embedded image reference in markdown.
+
+![Annotated procedure topic](../image/example-reference-topic.png)
+
+
+#### Tabs
+
+Place multiple versions of the same procedure (such as different versions for different products)
```
Comment (snneji) [inline template-procedure.md:217]: I think a better example here would be using tabs for kn CLI procedure vs the YAML procedure instead of multiple versions.

File: template-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Troubleshooting template
+
+When writing guidance to help to troubleshoot specific errors, the error must include:
+
```
Comment (snneji) [inline template-troubleshooting.md:4]: Maybe also have guidance on the Error Description here, ie how long it should be, what it should include, what it should not include etc.

**Review by snneji — CHANGES_REQUESTED** [review]
A few suggestions here, mostly formatting and wording tweaks.

**Comment by snneji** [comment]
Fixes #3477

**Review by julz — CHANGES_REQUESTED** [review]
Same comment as on #3449 that it'd be really nice to use real PR/commit titles, even if they're just copied from the issue title (otherwise the git history will become really hard to follow)

File: template-procedure.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+1. [Name of Task 1 (for example, Apply default configuration)](#task-1)
+1. [(Optional) Name of Task 2](#task-2)
+
+!!! note
+    Unless the number of tasks in the procedure is particularly high, do not use numbered lead-ins in the task headings.
+    For example, instead of "Task 1: Apply default configuration", use "Apply default configuration".
+
+
+## Prerequisites
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-procedure.md:39]: I want to change because "Prerequisites" outnumbers "Before you begin" in the Knative docs.

File: template-concept.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+Use this template when writing conceptual topics.
+Conceptual topics explain how things work or what things mean.
+They provide helpful context to readers. They do not include procedures.
+
+## Template
+
+The template below includes the standard sections that should appear in conceptual topics, including a topic introduction sentence, an overview, and placeholders for additional sections and subsections.
+Copy and paste the markdown from the template to use it in your topic.
+
+This topic describes...
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-concept.md:12]: Ehhh, maybe? It would look better in preview (not something I can get working for this doc, though) but engineers will just be looking at this in their text editors so I don't know that code tags would be more help than nuisance.

File: template-procedure.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+to perform a task as well as some context about the task.
+
+
+## Template
+
+The template below includes the standard sections that should appear in procedural topics, including
+a topic sentence, an overview section, and sections for each task within the procedure.
+Copy and paste the markdown from the template to use it in your topic.
+
+This topic describes...
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-procedure.md:13]: Ehhh, maybe? It would look better in preview (not something I can get working for this doc, though) but engineers will just be looking at this in their text editors so I don't know that code tags would be more help than nuisance.

File: template-procedure.md (hunk 2 above)
Comment (RichardJJG) [reply template-procedure.md:220]: Thanks, well caught.

File: template-procedure.md (hunk 3 above)
Comment (RichardJJG) [reply template-procedure.md:217]: Yep, I'll change that.

File: template-procedure.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+Write a few sentences to describe the task and provide additional context on the task.
+
+[Task]:
+
+1. Step 1
+2. Step 2
+
+
+## Procedure Content Samples
+
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-procedure.md:104]: Okay.

File: template-concept.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+Write a sentence or two to describe the content in this section.
+
+### Subsection Title
+
+Write a sentence or two to describe the content in this section.
+
+
+## Conceptual Content Samples
+
+The following are common content types that may appear in conceptual topics.
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-concept.md:38]: Will do

File: template-procedure.md (hunk 1 above)
Comment (RichardJJG) [reply template-procedure.md:212]: Yep, will do.

File: template-concept.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-concept.md:12]: I'll add the code tag; it's likely more help than hindrance.

File: template-concept.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+Do not skip header levels: H2 >> H3, not H2 >> H4.
+
+### Subsection Title
+
+Write a sentence or two to describe the content in this section.
+
+### Subsection Title
+
+Write a sentence or two to describe the content in this section.
+
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-concept.md:34]: Yep, will do.

File: template-procedure.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-procedure.md:13]: I'll add the code tag; it's likely more help than hindrance.

File: template-procedure.md [hunk 11]
```diff
@@ -0,0 +1,251 @@
...
+[Task]:
+
+1. Step 1
+1. Step 2
+
+
+## (Optional) Task 2
+
+Write a few sentences to describe the task and provide additional context on the task.
```
[the remark this answers is not in the document]
Comment (RichardJJG) [reply template-procedure.md:95]: Thanks, will do.

**Comment by RichardJJG** [comment]
Could this be merged now, @julz, @snneji? (Maybe at least one of you has permission to merge?)

File: template-concept.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (snneji) [reply template-concept.md:13]: I think the code snippets will show where the template begins. I don't think tags will hinder anything. I found it difficult to see where the introduction to the section ended and the template began before the code tags were added.

**Comment by RichardJJG** [comment]
Could you whack an /approved on this, please, @julz?

**Review by julz — APPROVED** [review]
/approve

**Comment by RichardJJG** [comment]
Ah, do you happen to know why this is failing "pull-knative-docs-integration-tests", @julz? 
I've looked at the details but they don't tell me which parts of the content is causing upset (if any).

**Comment by julz** [comment]
hm, not sure, let's try rerunning and see if that improves anything

/retest

**Comment by RichardJJG** [comment]
@julz, thanks -- seems that did the trick!
